Update language for user profile and user card

laurynmm opened the pull request:
Updates the web-app and documentation for user-facing uses of:
- "profile summary" to be "user card"
- "full profile" to be "profile"

Fixes #23530.

---

**Notes**:
- The 1st commit focuses on the "user card" update. The 2nd commit focuses on the "profile" update. They can be squashed before merging.
- In the issue description, there was a reference to "profile card" when discussing updating the help center references, which I assumed was a typo and that we wanted to use "user card" as stated at the top of the issue.
- This pull request focuses on user-facing updates. There are some follow-up items to consider:
  - The server/backend code has references to "profile summary" as part of some json error responses that is not updated by these changes. And the CustomProfileField model has a `display_in_profile_summary` field as well.
  - The frontend code also has various references to `display_in_profile_summary`

---

**Screenshots and screen captures:**

<details>
<summary>Web app updates</summary>

### Me message hover
![Screenshot from 2022-11-14 18-32-24](https://user-images.githubusercontent.com/63245456/201727921-d973a7bd-be36-4a63-b100-7862c7bd2328.png)
![Screenshot from 2022-11-14 18-32-28](https://user-images.githubusercontent.com/63245456/201727922-a9c262ed-d4cd-43a4-8b08-6428b6ba0c6e.png)

### Normal message hover
![Screenshot from 2022-11-14 18-32-32](https://user-images.githubusercontent.com/63245456/201727927-600df0b7-f2d1-433a-ba08-743013b69f2e.png)
![Screenshot from 2022-11-14 18-32-35](https://user-images.githubusercontent.com/63245456/201727929-0530092d-b555-4cb5-a4ea-f94beac225c9.png)

### Keyboard help menu
![Screenshot from 2022-11-14 18-33-18](https://user-images.githubusercontent.com/63245456/201727935-742d455b-337e-495a-b4b5-c91273861475.png)

### Custom profile org settings
![Screenshot from 2022-11-14 18-33-58](https://user-images.githubusercontent.com/63245456/201727939-fb82660c-11a0-435d-8152-5b400cd2f286.png)

### User card
![Screenshot from 2022-11-14 18-34-44](https://user-images.githubusercontent.com/63245456/201727942-015025f6-2f2e-40b1-8afb-91e5114a7102.png)

### Edit custom profile field modal
![Screenshot from 2022-11-14 18-35-26](https://user-images.githubusercontent.com/63245456/201727947-47791e73-6673-4f43-8eb2-23b4a1d156be.png)

### Add custom profile field modal
![Screenshot from 2022-11-14 18-35-37](https://user-images.githubusercontent.com/63245456/201727953-04274e7a-3816-4b44-8e85-7b6546897244.png)
</details>

<details>
<summary>Help center updates</summary>

### Custom profile fields
[Current documentation](https://zulip.com/help/custom-profile-fields#display-custom-fields-in-user-profile-summaries)
![Screenshot from 2022-11-14 18-43-01](https://user-images.githubusercontent.com/63245456/201729420-fb4045a7-29ba-4e8a-893a-d84b78078bdc.png)

### Keyboard shortcuts
[Current documentation](https://zulip.com/help/keyboard-shortcuts#message-actions)
![Screenshot from 2022-11-14 18-44-48](https://user-images.githubusercontent.com/63245456/201729752-4207569b-6137-4c3b-8a45-373f94328d69.png)

### Add or remove users from stream
[Current documentation](https://zulip.com/help/add-or-remove-users-from-a-stream#from-a-users-profile-alternate-method)
![Screenshot from 2022-11-14 18-46-46](https://user-images.githubusercontent.com/63245456/201730280-d109075b-d388-4dec-9ba7-2090d54b3016.png)

### Change a user's role
[Current documentation](https://zulip.com/help/change-a-users-role#change-a-users-role_1)
![Screenshot from 2022-11-14 18-48-42](https://user-images.githubusercontent.com/63245456/201730635-642127a6-f5ba-4c12-ae20-e541fcbd68ac.png)

### View messages sent by a user
[Current documentation](https://zulip.com/help/view-messages-sent-by-a-user)
![Screenshot from 2022-11-14 18-51-11](https://user-images.githubusercontent.com/63245456/201731053-f22071cf-510a-4ece-8021-165514a0b9dc.png)


</details>

<details>
<summary>API docs updates - Register queue</summary>

![Screenshot from 2022-11-14 18-42-03](https://user-images.githubusercontent.com/63245456/201729180-6e71f8b1-1a50-4b98-bb76-09cc696fe6a7.png)
![Screenshot from 2022-11-14 18-42-20](https://user-images.githubusercontent.com/63245456/201729182-d760dc78-2a23-4169-93a8-f87385955433.png)

</details>

---

**Self-review checklist**

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [x] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/version-control.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.

alya commented:
> In the issue description, there was a reference to "profile card" when discussing updating the help center references, which I assumed was a typo and that we wanted to use "user card" as stated at the top of the issue.

Oops, yes, that's a typo. Fixed.

alya commented:
All the screenshots look great to me! I'm poking around the help center to see if anything else can be improved. So far, I found.

On /help/private-messages, do you think this would be better? It's a bit of a tangly sentence.

> You can also click on any user in the right sidebar, or click on a user's profile picture or name, and select **Send private message**.
--> 
> You can also click on any user in the right sidebar, or click on a user's profile picture or name, to open their **user card**, and select **Send private message**.

alya commented:
Oh, I think the "Via the user's profile" tab should be named "Via user card" (or something like that), since it does not (currently, thought we plan to change this) involve opening the user's profile (in the new terminology).

alya commented:
On /help/status-and-availability:

> You can view status messages by hovering over your or anyone else's name in the left or right sidebar, or by clicking the user's name or avatar in the main message feed. 
-->
> You can view status messages by hovering over your or anyone else's name in the left or right sidebar, or by clicking the user's name or avatar in the main message feed to open their **user card**.

laurynmm commented:
> On /help/private-messages, do you think this would be better? It's a bit of a tangly sentence.
> 
> > You can also click on any user in the right sidebar, or click on a user's profile picture or name, and select **Send private message**.
> > -->
> > You can also click on any user in the right sidebar, or click on a user's profile picture or name, to open their **user card**, and select **Send private message**.

Here's your update as a screenshot ...

<details>
<summary>Alya's update</summary>

![Screenshot from 2022-11-14 19-38-47](https://user-images.githubusercontent.com/63245456/201739761-a520f353-b9bd-44f2-aeec-c87f354453f9.png)
</details>

I got a bit confused since it's two separate possible actions, so I made a version with two sentences ...

<details>
<summary>Lauryn's update</summary>

![Screenshot from 2022-11-14 19-45-38](https://user-images.githubusercontent.com/63245456/201741192-e794ac4b-9965-42e8-889e-0619eb86cd5d.png)
</details>

alya commented:
Maybe something like this?

> You can also click on any user in the right sidebar to start composing a private message to them. Or open their **user card** by clicking on their profile picture or name, and select **Send private message**.

alya commented:
On help/mention-a-user-or-group, the "Via a message" instructions should also be tweaked:

> Click on a user's profile picture or name on a message they sent.
->
> Click on a user's profile picture or name on a message they sent to open their user card.

laurynmm commented:
**Updates**:
- Added a commit to update the tabbed section for "Via the user's profile" to be "Via user card". See first screenshot below.
- Updated references in help center where we mention clicking on a user's name, avatar or profile picture in a message view opens the **user card**. See updated screenshots below.

*All the web-app and API documentation screenshots above are all still accurate as of these updates*.

---

**Screenshots**:

<details>
<summary>Updated "Via user card" tab example</summary>

![Screenshot from 2022-11-14 20-08-27](https://user-images.githubusercontent.com/63245456/201747800-7d95926f-1bf2-440c-a9a0-9a9e1dfa41f9.png)
</details>
<details>
<summary>Update to "Mention a user or group" via message</summary>

[Current documentation](https://zulip.com/help/mention-a-user-or-group#via-a-message)
![Screenshot from 2022-11-14 20-14-15](https://user-images.githubusercontent.com/63245456/201747804-b561ed76-b172-4fc1-9193-78003c186164.png)
</details>
<details>
<summary>View someone's profile tip update</summary>

[Current documentation](https://zulip.com/help/view-someones-profile#view-someones-profile_1)
![Screenshot from 2022-11-14 20-15-21](https://user-images.githubusercontent.com/63245456/201747812-1c0f3b4f-9295-4e6e-96a6-206e1457c6a1.png)
</details>
<details>
<summary>Mute a user tip update</summary>

[Current documentation](https://zulip.com/help/mute-a-user#mute-a-user_1)
![Screenshot from 2022-11-14 20-17-41](https://user-images.githubusercontent.com/63245456/201747813-44da346c-deb0-41cc-961a-cba7529a2af7.png)
</details>
<details>
<summary>Private messages tip update</summary>

[Current documentation](https://zulip.com/help/private-messages#send-a-pm)
![Screenshot from 2022-11-14 20-30-28](https://user-images.githubusercontent.com/63245456/201748784-74bcb97f-ed61-47b5-a85e-22342453fb12.png)
</details>
<details>
<summary>View status update</summary>

[Current documentation](https://zulip.com/help/status-and-availability#view-a-status)
![Screenshot from 2022-11-14 20-33-39](https://user-images.githubusercontent.com/63245456/201749378-55f03b14-ec49-4e84-8878-3eb763cbf023.png)
</details>

alya commented:
Looks great to me -- thanks for the quick iteration on this!

@timabbott tagging for your review.

timabbott commented:
This is great, merged after a small tweak in the comment messages to start with `popovers: Rename ...`. 

Thanks for making this happen quickly @laurynmm!
